samples/Mvc.LocalizationSample.Web/MaxLengthTwoAttribute.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+
+namespace Mvc.LocalizationSample.Web
+{
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter,
+        AllowMultiple = false)]
+    public class MaxLengthTwoAttribute : SelfContainedValidationAttribute
+    {
+        private const string _errorMessage = "MaxLengthTwo";
+
+        public MaxLengthTwoAttribute()
+        {
+            ErrorMessage = _errorMessage;
+        }
+
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            var result = base.IsValid(value, validationContext);
+
+            if (result.ErrorMessage != null)
```

Comment:
I had explicitly done `!= null` because I thought it was valid for someone to set the ErrorMessage to the empty string in their attribute, but I can't find anything to support that, so sure.